ui/js/graphs.js

```diff
@@ -705,18 +705,22 @@ perf.controller('GraphsCtrl', [
       });
   }]);
 
-perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
-                                            projects, optionCollectionMap,
-                                            timeRange, thServiceDomain,
-                                            PhSeries, defaultProjectName,
-                                            defaultPlatform) {
+perf.controller('TestChooserCtrl', function TestChooserCtrl($scope, $modalInstance, $http, 
+               projects, optionCollectionMap, timeRange, thServiceDomain, ThRepositoryModel, 
+               thDefaultRepo, PhSeries, defaultProjectName, defaultPlatform) {
   $scope.timeRange = timeRange;
   $scope.projects = projects;
   if (defaultProjectName) {
```

Comment:
Instead of this, I would so something like:

var projectName = defaultProjectName ? defaultProjectName : thDefaultRepo;

That removes the need for a bunch of if/else clauses.